src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,63 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  query: string;
+  rule: string;
+  setQuery: (v: string) => void;
+  chooseRule: (v: string) => void;
```

Comment:
What is `v`? The name should be descriptive, and as I mentioned previously the filter should have corresponding type